src/DOM/DOM.js

```diff
@@ -0,0 +1,188 @@
+ * @param {*} options
+ */
+function DOMEvent (type, listener, options) {
+	document.addEventListener(type, listener, options)
+}
+
+/**
+ * @return {Node}
+ */
+function DOMRoot () {
+	return document.documentElement
+}
+
+/**
+ * @param {Element} element
+ * @return {DOM}
+ */
+function DOMElement (element) {
+	if (element.xmlns) 
+		return DOM(document.createElementNS(element.xmlns, element.type))
+
+	return DOM(document.createElement(element.type))
+}
+
+/**
+ * @param {Element} element
+ * @return {DOM}
+ */
+function DOMText (element) {
+	return DOM(document.createTextNode(element.children))
+}
+
+/**
+ * @param {Element} element
+ */
+function DOMContent (element) {
+	DOMNode(element).textContent = ''
+}
+
+/**
+ * @param {Element} element
+ * @param {(string|number)} value
+ */
+function DOMValue (element, value) {
+	DOMNode(element).nodeValue = value
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ */
+function DOMRemove (element, parent) {
+	DOMNode(parent).removeChild(DOMNode(element))
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} parent
+ */
+function DOMInsert (element, sibling, parent) {
+	DOMNode(parent).insertBefore(DOMNode(element), DOMNode(sibling))
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ */
+function DOMAppend (element, parent) {
+	DOMNode(parent).appendChild(DOMNode(element))
+}
+
+/**
+ * @param {Element} element
+ * @param {string} name
+ * @param {*} value
+ */
+function DOMStyle (element, value) {
+	for (var key in value)
+		if (key.indexOf('-') < 0)
+			DOMNode(element).style[key] = value[key]
+		else
+			DOMNode(element).style.setProperty(key, value[key])
+}
+
+/**
+ * @param {Element} element
+ * @param {string} name 
+ * @param {*} value
+ */
+function DOMProperty (element, name, value) {
+	DOMNode(element)[name] = value
+}
+
+/**
+ * @param {Element} element
+ * @param {string} name
+ * @param {*} value
+ * @param {string} xmlns
+ * @param {string} key
+ */
+function DOMAttribute (element, name, value, xmlns, key) {
+	xmlns ? DOMNode(element)[key+'NS'](xmlns, name, value) : DOMNode(element)[key](name, value)
+}
+
+/**
+ * @param {Element} element
+ * @param {string} name
+ * @param {*} value
+ * @param {string} xmlns
+ */
+function DOMProperties (element, name, value, xmlns) {
+	switch (name) {
+		case 'xlink:href':
+			if (!xmlns)
+				return DOMProperties(element, name, value, 'http://www.w3.org/1999/xlink')
+		case 'key':
+		case 'xmlns':
+		case 'children':
+			return
+		case 'dangerouslySetInnerHTML':
+			return DOMProperties(element, 'innerHTML', value && value.__html, '')
+		case 'style':
+			if (typeof value === 'object')
+				return DOMStyle(element, value)
+			break
+		case 'className':
+			if (xmlns || value == null)
+				return DOMProperties(element, 'class', value, xmlns)
+		case 'id':
+			return DOMProperty(element, name, value)
+		case 'width':
+		case 'height':
+			if (element.type === 'img')
```

Comment:
I'm unfamiliar with this, `<img>` `height/width` must be set as attributes instead of properties?

Reply:
`setAttribute` allows the value to support units i.e px/em/etc while setting it as a property would not. i.e

```js
img.setAttribute('width', '100%')
// would calc what '100%' means and set that px value of that to the image, while
img.width = '100%'
// might not set it or, might just set it to 100px
```

Reply:
Because other DOM nodes do note have the a width/height property at all or do not have a width/height property that works like it does with `<img>`.

Reply:
for posterity, percent widths on `<img>` is deprecated in HTML5
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img

Reply:
I guess my point above about it supporting `em/etc` is invalid, it's just percent that it supports, but i guess that's the core of the issue since also setting `img.width = "100px"` would also fail, which is why it always falls back to using setAttribute for `<img>`.